backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceDAO.java

```diff
@@ -638,4 +639,16 @@ private boolean matchesDocType(@NonNull final I_C_Invoice serviceFeeInvoiceCandi
 
 		return docBaseAndSubType.equals(targetDocType);
 	}
+
+	public Collection<String> retrievePaidInvoiceDocNosForFilter(@NonNull final IQueryFilter<org.compiere.model.I_C_Invoice> filter)
+	{
+		return queryBL.createQueryBuilder(org.compiere.model.I_C_Invoice.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_Invoice.COLUMNNAME_IsPaid, true)
+				.addFilter(filter)
+				.create()
+				.stream()
+				.map(org.compiere.model.I_C_Invoice::getDocumentNo)
```

Comment:
better have a limit here..idk, maybe 20?..otherwise, if the user selects all invoices, this might blow the heap

Reply:
10 it is then.